src/EFCore.PG/Metadata/PostgresEnum.cs

```diff
@@ -9,102 +9,126 @@
-            [CanBeNull] string schema,
-            [NotNull] string name)
-        {
-            Check.NotNull(annotatable, nameof(annotatable));
-            Check.NotEmpty(name, nameof(name));
-
-            var annotationName = BuildAnnotationName(schema, name);
-
-            return annotatable[annotationName] == null ? null : new PostgresEnum(annotatable, annotationName);
-        }
-
-        static string BuildAnnotationName(string schema, string name)
-            => NpgsqlAnnotationNames.EnumPrefix + (schema == null ? name : schema + '.' + name);
-
-        public static IEnumerable<PostgresEnum> GetPostgresEnums([NotNull] IAnnotatable annotatable)
         {
             Check.NotNull(annotatable, nameof(annotatable));
+            Check.NullButNotEmpty(schema, nameof(schema));
+            Check.NotNull(name, nameof(name));
+            Check.NotNull(labels, nameof(labels));
 
-            return annotatable.GetAnnotations()
-                .Where(a => a.Name.StartsWith(NpgsqlAnnotationNames.EnumPrefix, StringComparison.Ordinal))
-                .Select(a => new PostgresEnum(annotatable, a.Name));
-        }
-
-        public Annotatable Annotatable => (Annotatable)_annotatable;
+            var annotationName =
+                schema != null
+                    ? $"{NpgsqlAnnotationNames.EnumPrefix}{schema}.{name}"
+                    : $"{NpgsqlAnnotationNames.EnumPrefix}{name}";
 
-        public string Schema => GetData().Schema;
+            var postgresEnum = new PostgresEnum(annotatable, annotationName);
 
-        public string Name => GetData().Name;
+            if (postgresEnum.Annotation == null)
+                postgresEnum.Labels = labels;
 
-        public string[] Labels
-        {
-            get => GetData().Labels;
-            set => SetData(value);
+            return postgresEnum;
         }
 
-        (string Schema, string Name, string[] Labels) GetData()
+        /// <summary>
+        /// Gets the collection of <see cref="PostgresEnum"/> stored in the <see cref="IAnnotatable"/>.
+        /// </summary>
+        /// <param name="annotatable">The annotatable to search for <see cref="PostgresEnum"/> annotations.</param>
+        /// <returns>
+        /// The collection of <see cref="PostgresEnum"/> stored in the <see cref="IAnnotatable"/>.
+        /// </returns>
+        /// <exception cref="ArgumentNullException"><paramref name="annotatable"/></exception>
+        [NotNull]
+        [ItemNotNull]
+        public static IEnumerable<PostgresEnum> GetPostgresEnums([NotNull] IAnnotatable annotatable)
+            => Check.NotNull(annotatable, nameof(annotatable))
+                    .GetAnnotations()
+                    .Where(a => a.Name.StartsWith(NpgsqlAnnotationNames.EnumPrefix, StringComparison.Ordinal))
+                    .Select(a => new PostgresEnum(annotatable, a.Name));
+
+        [CanBeNull]
+        IAnnotation Annotation => _annotatable.FindAnnotation(_annotationName);
```

Comment:
I'd suggest reverting this, aligning ourselves with the EF Core practice here. Properties such as `Name` can simply call `GetData()` as before, obviating the need for a private `Annotation` property. It's about keeping the code as similar as possible to how things are done in EF Core, unless there's a compelling reason to do otherwise.

Reply:
As below, this mattered more in a previous iteration of this PR. Will revert.